### DISCUSSION TITLE
How we can add a support for new Seializers gems?

### DISCUSSION THREAD

**Discussion by PedroAugustoRamalhoDuarte** [ElMassimo/types_from_serializers#11] [Ideas]
These gems help reduce a lot of the overhead of writing types twice in apps that use Rails and TypeScript. However, there are many other serializer gems. Is there an easy way we can build an adapter to make it simpler to support new serializer gems? For example, [active_model_serializers](https://github.com/rails-api/active_model_serializers) has a very similar API to OJ, but we cannot set it up with AMS right now.

**Comment by mmm-zesty** [comment]
I think this would be great. I have been using alba and migrating it over might not get adoption from the team.

**Comment by PedroAugustoRamalhoDuarte** [comment]
I am building a small POC to test it out. I think we can just override some methods from OjSerializer and make it work

**Reply by PedroAugustoRamalhoDuarte** [reply]
@ouichef you can do something like that to works with alba: https://github.com/ElMassimo/types_from_serializers/pull/18
